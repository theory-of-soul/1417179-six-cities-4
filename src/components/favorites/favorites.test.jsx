import React from "react";
import renderer from 'react-test-renderer';
import Favorites from "./favorites";
import {BrowserRouter} from "react-router-dom";

const offers = [{
  id: 1,
  img: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
  value: 235,
  time: `night`,
  isInBookmark: true,
  rating: 2.4,
  name: `The house among olive `,
  type: `apartment`,
  city: `Dusseldorf`,
  point: [51.237402, 6.797314],
  cityLocation: [51.225402, 6.776314],
  bedrooms: 4,
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  goods: [`Laptop friendly workspace`],
  guests: 6,
  host: {
    hostId: 25,
    name: `Angelina`,
    isSuper: true,
    icon: `img/avatar-angelina.jpg`
  },
  images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`],
  isPremium: false,
}];

describe(`Favorites component snapshot tests`, () => {
  it(`Favorites component without places`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Favorites
              favorites={{
                hasFavorites: false,
                offers: []
              }}
              onLoadFavorites={() => {}}
              addToFavoritesHandler={() => {}}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Favorites component with places`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Favorites
              favorites={{
                hasFavorites: true,
                offers: [{
                  city: `Dusseldorf`,
                  places: offers
                }]
              }}
              onLoadFavorites={() => {}}
              addToFavoritesHandler={() => {}}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
