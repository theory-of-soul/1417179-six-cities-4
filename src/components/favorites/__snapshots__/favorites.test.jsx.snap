// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component snapshot tests Favorites component with places 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites"
  >
    <h1
      className="favorites__title"
    >
      Saved listing
    </h1>
    <ul
      className="favorites__list"
    >
      <li
        className="favorites__locations-items"
      >
        <div
          className="favorites__locations locations locations--current"
        >
          <div
            className="locations__item"
          >
            <a
              className="locations__item-link"
              href="#"
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </div>
        </div>
        <div
          className="favorites__places"
        >
          <article
            className="place-card favorites__card"
            onMouseEnter={[Function]}
          >
            <div
              className="place-card__image-wrapper favorites__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="The house among olive "
                  className="place-card__image"
                  height={110}
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
                  width={150}
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    235
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / 
                    night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "48%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/1"
                  onClick={[Function]}
                >
                  The house among olive 
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
        </div>
      </li>
    </ul>
  </section>
</div>
`;

exports[`Favorites component snapshot tests Favorites component without places 1`] = `
<div
  className="page__favorites-container container"
>
  <section
    className="favorites favorites--empty"
  >
    <h1
      className="visually-hidden"
    >
      Favorites (empty)
    </h1>
    <div
      className="favorites__status-wrapper"
    >
      <b
        className="favorites__status"
      >
        Nothing yet saved.
      </b>
      <p
        className="favorites__status-description"
      >
        Save properties to narrow down search or plan yor future trips.
      </p>
    </div>
  </section>
</div>
`;
